File: javascript/fiscor-pdf-api-example.js
const baseUrl = "https://pdf.fiscor.am/api/v1";

export async function convertPdfToWord({ apiKey, file }) {
  const form = new FormData();
  form.append("file", file, file.name);

  const createResponse = await fetch(`${baseUrl}/jobs/pdf-to-word`, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${apiKey}`,
    },
    body: form,
  });

  const createBody = await createResponse.json();
  if (!createResponse.ok) {
    throw new Error(createBody.error ?? "Could not create conversion job.");
  }

  const jobId = createBody.jobId;
  await waitForCompletion(apiKey, jobId);

  const downloadResponse = await fetch(`${baseUrl}/jobs/${jobId}/download`, {
    headers: {
      Authorization: `Bearer ${apiKey}`,
    },
  });

  if (!downloadResponse.ok) {
    throw new Error("Could not download conversion result.");
  }

  return {
    jobId,
    blob: await downloadResponse.blob(),
  };
}

async function waitForCompletion(apiKey, jobId) {
  for (let attempt = 1; attempt <= 60; attempt++) {
    const response = await fetch(`${baseUrl}/jobs/${jobId}`, {
      headers: {
        Authorization: `Bearer ${apiKey}`,
      },
    });

    const body = await response.json();
    if (!response.ok) {
      throw new Error(body.error ?? "Could not read job status.");
    }

    if (body.status === "Completed") {
      return;
    }

    if (body.status === "Failed") {
      throw new Error(body.failureReason ?? "Conversion job failed.");
    }

    await delay(Math.min(2 + attempt, 10) * 1000);
  }

  throw new Error(`Job ${jobId} did not complete in time.`);
}

function delay(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

// Browser usage:
//
// const { blob } = await convertPdfToWord({
//   apiKey: "YOUR_API_KEY",
//   file: document.querySelector("input[type=file]").files[0],
// });
//
// const link = document.createElement("a");
// link.href = URL.createObjectURL(blob);
// link.download = "output.docx";
// link.click();

File: csharp/FiscorPdfApiExample.cs
using System.Net.Http.Headers;
using System.Text.Json;

var apiKey = Environment.GetEnvironmentVariable("FISCOR_PDF_API_KEY")
             ?? throw new InvalidOperationException("Set FISCOR_PDF_API_KEY.");
var baseUrl = Environment.GetEnvironmentVariable("FISCOR_PDF_BASE_URL")
              ?? "https://pdf.fiscor.am/api/v1";

using var http = new HttpClient { BaseAddress = new Uri(baseUrl.TrimEnd('/') + "/") };
http.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", apiKey);

var inputPath = args.Length > 0 ? args[0] : "input.pdf";
var outputPath = args.Length > 1 ? args[1] : "output.docx";

var jobId = await CreatePdfToWordJobAsync(http, inputPath);
Console.WriteLine($"Created job: {jobId}");

await WaitForCompletionAsync(http, jobId);
await DownloadResultAsync(http, jobId, outputPath);

Console.WriteLine($"Saved result: {outputPath}");

static async Task<string> CreatePdfToWordJobAsync(HttpClient http, string inputPath)
{
    await using var fileStream = File.OpenRead(inputPath);
    using var form = new MultipartFormDataContent();
    using var fileContent = new StreamContent(fileStream);
    fileContent.Headers.ContentType = new MediaTypeHeaderValue("application/pdf");
    form.Add(fileContent, "file", Path.GetFileName(inputPath));

    using var response = await http.PostAsync("jobs/pdf-to-word", form);
    var body = await response.Content.ReadAsStringAsync();
    response.EnsureSuccessStatusCode();

    using var json = JsonDocument.Parse(body);
    return json.RootElement.GetProperty("jobId").GetString()
           ?? throw new InvalidOperationException("Response did not include jobId.");
}

static async Task WaitForCompletionAsync(HttpClient http, string jobId)
{
    for (var attempt = 1; attempt <= 60; attempt++)
    {
        using var response = await http.GetAsync($"jobs/{jobId}");
        var body = await response.Content.ReadAsStringAsync();
        response.EnsureSuccessStatusCode();

        using var json = JsonDocument.Parse(body);
        var status = json.RootElement.GetProperty("status").GetString();

        if (string.Equals(status, "Completed", StringComparison.OrdinalIgnoreCase))
        {
            return;
        }

        if (string.Equals(status, "Failed", StringComparison.OrdinalIgnoreCase))
        {
            throw new InvalidOperationException($"Job failed: {body}");
        }

        await Task.Delay(TimeSpan.FromSeconds(Math.Min(2 + attempt, 10)));
    }

    throw new TimeoutException($"Job {jobId} did not complete in time.");
}

static async Task DownloadResultAsync(HttpClient http, string jobId, string outputPath)
{
    using var response = await http.GetAsync($"jobs/{jobId}/download");
    response.EnsureSuccessStatusCode();

    await using var output = File.Create(outputPath);
    await response.Content.CopyToAsync(output);
}
